hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionHealthProblem.java

```diff
@@ -0,0 +1,116 @@
+
+        @Override
+        public boolean occurs(SubscriptionHealthContext context) {
+            SubscriptionMetrics subscriptionMetrics = context.subscriptionMetrics;
+            return subscriptionMetrics.otherErrorsRate > MAX_OTHER_ERRORS_RATIO * subscriptionMetrics.rate;
+        }
+    },
+
+    TIMING_OUT {
+        private static final double MAX_TIMEOUTS_RATIO = 0.1;
+
+        @Override
+        public boolean occurs(SubscriptionHealthContext context) {
+            SubscriptionMetrics subscriptionMetrics = context.subscriptionMetrics;
+            return subscriptionMetrics.timeoutsRate > MAX_TIMEOUTS_RATIO * subscriptionMetrics.rate;
+        }
+    },
+
+    MALFUNCTIONING {
+        private static final double MAX_5XX_ERRORS_RATIO = 0.1;
+
+        @Override
+        public boolean occurs(SubscriptionHealthContext context) {
+            SubscriptionMetrics subscriptionMetrics = context.subscriptionMetrics;
+            return subscriptionMetrics.code5xxErrorsRate > MAX_5XX_ERRORS_RATIO * subscriptionMetrics.rate;
+
+        }
+    },
+
+    RECEIVING_MALFORMED_MESSAGES {
+        private static final double MAX_4XX_ERRORS_RATIO = 0.1;
+
+        @Override
+        public boolean occurs(SubscriptionHealthContext context) {
+            Subscription subscription = context.subscription;
+            SubscriptionMetrics subscriptionMetrics = context.subscriptionMetrics;
+            return hasClientErrorRetry(subscription) && isCode4xxErrorRateHigh(subscriptionMetrics);
+        }
+
+        private boolean hasClientErrorRetry(Subscription subscription) {
+            if (subscription.isBatchSubscription()) {
+                return subscription.getBatchSubscriptionPolicy().isRetryClientErrors();
+            } else {
+                return subscription.getSerialSubscriptionPolicy().isRetryClientErrors();
+            }
+        }
+
+        private boolean isCode4xxErrorRateHigh(SubscriptionMetrics subscriptionMetrics) {
+            return subscriptionMetrics.code4xxErrorsRate > MAX_4XX_ERRORS_RATIO * subscriptionMetrics.rate;
+        }
+    };
+
+    public abstract boolean occurs(SubscriptionHealthContext context);
+
+    public static final class SubscriptionHealthContext {
+        private final Subscription subscription;
+        private final TopicMetrics topicMetrics;
+        private final SubscriptionMetrics subscriptionMetrics;
+
+        public SubscriptionHealthContext(Subscription subscription, TopicMetrics topicMetrics, SubscriptionMetrics subscriptionMetrics) {
+            this.subscription = subscription;
+            this.topicMetrics = topicMetrics;
+            this.subscriptionMetrics = subscriptionMetrics;
+        }
+    }
+
+    public static final class TopicMetrics {
+        private final double rate;
+
+        public TopicMetrics(double rate) {
+            this.rate = rate;
+        }
+    }
+
+    public static final class SubscriptionMetrics {
```

Comment:
Because their single use case is to pass metrics to SubscriptionHealthProblem.occurs(). I created these classes because the top-level SubscriptionMetrics and TopicMetrics contain String values which have to be parsed (and can be unparsable). I thought that it is the responsibility of SubscriptionHealthChecker to decide whether the subscription health is HEALTHY, UNHEALTHY or NO_DATA and the SubscriptionHealthProblems responsibility is to calculate if they occur.